inferno-vc/src/Inferno/VersionControl/Operations.hs

```diff
@@ -387,18 +427,30 @@ getAllHeads = do
       []
       (map takeFileName headsRaw)
 
-fetchFunctionsForGroups :: (VCStoreLogM env m, VCStoreErrM err m, VCStoreEnvM env m, Ord g, FromJSON a, FromJSON g) => Set.Set g -> m [VCMeta a g VCObjectHash]
+-- | Fetch all objects that are public or that belong to the given set of groups.
+-- Note this is a potentially long operation so no locks are held while traversing the
```

Comment:
I agree with this (at least for now). We are calling this function multiple times in Onping's selector. Lock will increase the respond time with each query.